test/xpath/test_base.rb

```diff
@@ -1158,5 +1159,15 @@ def test_or_and
       end
       assert_equal(["/"], hrefs, "Bug #3842 [ruby-core:32447]")
     end
+
+    def test_match_with_deprecated_usage
+      verbose, $VERBOSE = $VERBOSE, nil
+      doc = Document.new("<a><b/></a>")
+      assert_equal ['b'], XPath.match([doc, doc], '//b').map(&:name)
+      assert_equal ['b'], XPath.match([doc], '//b').map(&:name)
+      assert_equal [], XPath.match([], '//b').map(&:name)
```

Comment:
Could you avoid omitting parenthesis for assertions to avoid potential warnings?
See also: https://bugs.ruby-lang.org/issues/20922

```suggestion
      assert_equal(['b'], XPath.match([doc, doc], '//b').map(&:name))
      assert_equal(['b'], XPath.match([doc], '//b').map(&:name))
      assert_equal([], XPath.match([], '//b').map(&:name))
```